server/models/user.js

```diff
@@ -1,37 +1,152 @@
 const mongoose = require("mongoose");
 const Schema = mongoose.Schema;
 const validator = require('validator');
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+const keys = require('../config/keys');
+const List = require('./list');
+const Item = require('./item');
+
 
 
 const userSchema = new Schema({
     name: {
         type: String,
         required: true,
-        trim: true
+        trim: true,
+        min: [4, 'Too short, min is 4 characters']
     },
     email: {
         type: String,
         required: true,
         trim: true,
         unique: true,
+        lowercase: true,
         validate(value) {
-            if(!validator.isEmail(value)) throw new Error('Email is invalid.')
-        }
+            if(!validator.isEmail(value)) throw new Error('Email is invalid!')
+        },        
+        min: [4, 'Too short, min is 4 characters']
     },
     password: {
         type: String,
         required: true,
         trim: true,
-        minlength: 7,
+        min: [4, 'Too short, min is 4 characters'],
         validate(value) {
             if (value.toLowerCase().includes('password')) throw new Error('Password cannot contain "pasword". Plese try another one!')
         }
     },
+    tokens: [{
```

Comment:
Remove this before we merge this PR

Reply:
Hey aeciorc, I have already set tokens to be part of auth-middleware